cmd/agent/common/commands/check.go

```diff
@@ -81,6 +82,7 @@ func setupCmd(cmd *cobra.Command) {
 	cmd.Flags().BoolVarP(&saveFlare, "flare", "", false, "save check results to the log dir so it may be reported in a flare")
 	cmd.Flags().UintVarP(&discoveryTimeout, "discovery-timeout", "", 5, "max retry duration until Autodiscovery resolves the check template (in seconds)")
 	cmd.Flags().UintVarP(&discoveryRetryInterval, "discovery-retry-interval", "", 1, "duration between retries until Autodiscovery resolves the check template (in seconds)")
+	cmd.Flags().UintVarP(&discoveryInstances, "discovery-instances", "", 1, "number of check instances to wait (retry/wait until the specified number of instances is not reached)")
```

Comment:
```suggestion
	cmd.Flags().UintVarP(&discoveryInstances, "discovery-min-instances", "", 1, "minimum number of config instances to be discovered before running the check(s)")
```

No strong opinion :)